Clarify that whiteout filenames require a non-empty basename (#1314)
opencontainers/image-spec: layer.md

Implementations on Windows MUST support these additional attributes, encoded in [PAX vendor
extensions](https://github.com/libarchive/libarchive/wiki/ManPageTar5#pax-interchange-format) as follows:

> - [Windows file attributes](<https://msdn.microsoft.com/en-us/library/windows/desktop/gg258117(v=vs.85).aspx>) (`MSWINDOWS.fileattr`)

Reviewer: Is this change needed?

Author: not needed

Reviewer: ```suggestion
- [Windows file attributes](https://msdn.microsoft.com/en-us/library/windows/desktop/gg258117(v=vs.85).aspx) (`MSWINDOWS.fileattr`)
```

In that case, lets revert it for consistency.

- A whiteout file is an empty file with a special filename that signifies a path should be deleted.
- A whiteout filename consists of the prefix `.wh.` plus the basename of the path to be deleted.

> - A whiteout filename MUST include a non-empty basename after the `.wh.` prefix. A bare `.wh.` entry with no basename is invalid and implementations MUST return an error when encountering such an entry. To remove all children of a directory, use an opaque whiteout (`.wh..wh..opq`) as described below.

Reviewer: ```suggestion
- A `.wh.` file, without a basename to delete, is invalid and implementations SHOULD return an error when encountering such an entry.
```

I think the language should be softened to a SHOULD since not every implementation is a container runtime, e.g. a scanner looking for leaked credentials in the layer should not be blocked by a maliciously crafted layer. This can also be simplified to the one sentence.

Author: Makes sense, accepted the suggestion.
